doc/components/popup.md

```diff
@@ -0,0 +1,481 @@
+    enter = true,
+    focusable = true,
+    zindex = 10,
+    relative = "editor",
+    border = {
+        highlight = "MyHighlightGroup",
+        padding = {
+            top = 2,
+            bottom = 2,
+            left = 3,
+            right = 3,
+        },
+        style = "double",
+        text = {
+            top = " I am top title ",
+            top_align = "center",
+            bottom = "I am bottom title",
+            bottom_align = "left",
+        },
+    },
+    buf_options = {
+        modifiable = true,
+        readonly = false,
+    },
+    win_options = {
+        winblend = 10,
+        winhighlight = "Normal:Normal",
+    },
+)
+```
+
+## Property Descriptions
+
+### `border`
+
+- **Type:** `table`
+- **Description:** Contains all border related options
+
+#### `padding`
+
+- **Type:** `table`
+- **Description:**
+    - Controls the popup padding
+    - Behaves like [CSS padding](https://developer.mozilla.org/en-US/docs/Web/CSS/padding)
+
+##### Usage
+
+```lua
+border = {
+    padding = { top = 10, bottom = 20, left = 15, right = 20 }
+}
+```
+
+```lua
+border = {
+    padding = { 10, 30, 40, 10 }
+}
+```
+
+#### `highlight`
+
+- **Type:** `string`
+- **Description:** Highlight group name for the border characters.
+
+```lua
+border = {
+    highlight = "MyHighlightGroup"
+}
+```
+
+#### `style`
+
+- **Type:** `string` or `table`
+- **Description:** Controls the styling of the border
+
+##### Usage
+
+###### Predefined Layouts
+
+- Can be: `double`, `none`, `rounded`, `shadow`, `single`, `solid`
+
+```lua
+border = {
+    style = "double"
+}
+```
+
+###### List
+
+- Order is clockwise
+
+```lua
+border = {
+    style = { "╭", "─", "╮", "│", "╯", "─", "╰", "│" }
+}
+```
+
+###### List
+
+```lua
+border = {
+    style = {
+        top_left = "╭",
+        top = "─",
+        top_right = "╮",
+        left = "│",
+        right = "│",
+        bottom_left = "╰",
+        bottom = "─",
+        bottom_right = "╯",
+    }
+}
+```
+
+#### `text`
+
+- **Type:** `string` or `table`
+- **Description:**
+    - Text displayed in the border to serve as a title
+    - `top_align` and `bottom_align` can be: `left`, `right`, `center`
+
+##### Usage
+
+```lua
+border = {
+    text = {
+        top = "I am top title",
+        top_align = "center",
+        bottom = "I am bottom title",
+        bottom_align = "left",
+    }
+}
+```
+
+---
+
+### `relative`
+
+- **Type:** `string` or `table`
+- **Description:** This option affects how `position` and `size` is calculated.-
+
+##### Usage
+
+###### Cursor
+
+- Relative to cursor on current window
+
+```lua
+relative = "cursor"
+```
+
+###### Editor
+
+- Relative to the current editor on current window
+
+```lua
+relative = "editor"
+```
+
+###### Cursor (default)
+
+- Relative to the current window
+
+```lua
+relative = "win"
+```
+
+###### Target Window
+
+- Relative to the window of `id`
+
+```lua
+relative = {
+    type = "win",
+    winid = 5,
+}
+```
+
+###### Buffer position
```

Comment:
These headers are way too deep (6 levels) 😅 Since each component will have separate documentation file, I think we can keep markdown headers within 3 levels.

Reply:
Will address these by converting some of them to bolded text.